pkg/scheduler/scheduler_test.go

```diff
@@ -953,27 +953,19 @@ func TestSchedulerWithVolumeBinding(t *testing.T) {
 				t.Fatalf("scheduling timeout after %v", wait.ForeverTestTimeout)
 			}
 			stopFunc()
-			// Wait for scheduling to return an error
+			// Wait for scheduling to return an error or succeed binding.
 			select {
 			case err := <-errChan:
 				if item.expectError == nil || !reflect.DeepEqual(item.expectError.Error(), err.Error()) {
 					t.Errorf("err \nWANT=%+v,\nGOT=%+v", item.expectError, err)
 				}
-			case <-time.After(chanTimeout):
-				if item.expectError != nil {
-					t.Errorf("did not receive error after %v", chanTimeout)
-				}
-			}
-
-			// Wait for pod to succeed binding
-			select {
 			case b := <-bindingChan:
 				if !reflect.DeepEqual(item.expectPodBind, b) {
 					t.Errorf("err \nWANT=%+v,\nGOT=%+v", item.expectPodBind, b)
 				}
 			case <-time.After(chanTimeout):
 				if item.expectPodBind != nil {
-					t.Errorf("did not receive pod binding after %v", chanTimeout)
+					t.Errorf("did not receive pod binding or error after %v", chanTimeout)
```

Comment:
We should also error out below `if item.expectError != nil`.

Reply:
we don't actually need to check anything in the case of timeout. It's an error regardless.